src/Http/Routing/src/EndpointRoutingMiddleware.cs

```diff
@@ -55,12 +56,10 @@ internal sealed class EndpointRoutingMiddleware
 
         public Task Invoke(HttpContext httpContext)
         {
-            var feature = new EndpointSelectorContext(httpContext);
-
             // There's already an endpoint, skip maching completely
-            if (feature.Endpoint != null)
+            if (httpContext.GetEndpoint() != null)
```

Comment:
nit: Assign GetEndpoint to variable and reuse variable

Reply:
note that I'm still fixing 1.2 more thigns here.